File: Java/Polymorphism.java
package Java;

class Animal {
    void sound() {
        System.out.println("Animal sound");
    }
}

class Cat extends Animal {
    void sound() {
        System.out.println("Cat meows");
    }
}

class Dog extends Animal {
    void sound() {
        System.out.println("Dog barks");
    }
}

public class Polymorphism {
    public static void main(String[] args) {
        Animal a1 = new Cat();
        Animal a2 = new Dog();
        a1.sound();
        a2.sound();
    }
}

File: Dart/polymorphism.dart
class Animal {
  void sound() {
    print('Animal sound');
  }
}

class Cat extends Animal {
  @override
  void sound() {
    print('Cat meows');
  }
}

class Dog extends Animal {
  @override
  void sound() {
    print('Dog barks');
  }
}

void main() {
  Animal a1 = Cat();
  Animal a2 = Dog();
  a1.sound(); 
  a2.sound(); 
}